### PR TITLE
add my enumerable method implementations 

### PR DESCRIPTION
**Pull request by janessatran**
implemented: 

- my_each: iterates through each item in object
- my_each_with_index: iterates through object and returns item and index
- my_select: returns items in object that appear in search condition 
- my_all: returns true/false if all items meet condition

### CODE REVIEW

File: my_enumerable_module.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        if not yield item
+          flag = false
+          break
+        end
+      end
+    end
+    flag
+  end
+
+  def bubble_sort(arr)
```
Comment (slabounty) [inline my_enumerable_module.rb]: Did you really want to leave `bubble_sort` and `bubble_sort_by` in this file?

File: spec/my_enumerable_module_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+require ('./my_enumerable_module.rb')
+
+describe Enumerable do 
+
```
Comment (slabounty) [inline spec/my_enumerable_module_spec.rb:4]: Extra blank line.

File: spec/my_enumerable_module_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  describe '#my_each' do 
+
```
Comment (slabounty) [inline spec/my_enumerable_module_spec.rb:6]: Extra blank line.

File: spec/my_enumerable_module_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    end
+
+    it 'does nothing if not block is given' do
+      expect([].my_each).to eq([])
+    end
+
+  end
+
+  describe '#my_each_with_index' do
+
```
Comment (slabounty) [inline spec/my_enumerable_module_spec.rb:27]: Extra blank line.

File: spec/my_enumerable_module_spec.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    it 'returns each item and index in the object' do
+      expect(['hi', 'hello'].my_each_with_index {|val, idx| puts "index: #{idx} for val: #{val}" }).to eq(['hi', 'hello'])
+    end
+
+  end 
+
+  describe '#my_select' do 
+
```
Comment (slabounty) [inline spec/my_enumerable_module_spec.rb:35]: Extra blank line.

File: my_enumerable_module.rb [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        comparison = yield arr[idx], arr[idx+1]
+        if comparison > 0
+          arr[idx], arr[idx+1] = arr[idx+1], arr[idx]
+          swapped = true
+        end
+      end
+    end
+    arr
+  end
+
```
Comment (slabounty) [inline my_enumerable_module.rb:145]: Extra blank line.

File: my_enumerable_module.rb [hunk 7]
```diff
@@ -0,0 +1,72 @@
+module Enumerable
```
Comment (slabounty) [inline my_enumerable_module.rb:1]: I see this was part of the project, but please note, that opening an existing module or class this way and adding things while doable and sometimes a good idea, can often lead to a very bad place lol.

File: spec/my_enumerable_module_spec.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
+require ('./my_enumerable_module.rb')
+
+describe Enumerable do
```
Comment (slabounty) [inline spec/my_enumerable_module_spec.rb]: You should write some tests in here that *don't* use arrays. Think about the requirements for using enumerable and see if you can't implement that and then use your `my_xxx` methods.

File: spec/my_enumerable_module_spec.rb [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    end
+
+  end 
+
+  describe '#my_select' do 
+
+    it 'selects items out of the object that appear in the search regex' do
+      expect(['a', 'b', 'c', 'd', 'e'].my_select {|i| i =~ /[aeiou]/}).to eq(['a', 'e'])
+    end
+
```
Comment (slabounty) [inline spec/my_enumerable_module_spec.rb:39]: Extra blank line.

File: my_enumerable_module.rb (hunk 7 above)
Comment (janessatran) [reply my_enumerable_module.rb:1]: lol good to know! do you have suggestions for what i could change it to? maybe i can do something like 

```
class MyEnumerable
  include Enumerable 
```
but that would require me to also change all my method names right?

File: spec/my_enumerable_module_spec.rb (hunk 8 above)
Comment (janessatran) [reply spec/my_enumerable_module_spec.rb]: okay! thanks for the suggestion!

File: spec/my_enumerable_module_spec.rb (hunk 9 above)
Comment (janessatran) [reply spec/my_enumerable_module_spec.rb:39]: is there a way to easily spot blank lines? i have a lot of them and highlighting all the text in my editor doesn't seem to be a clear way for me to spot them.. hmm..

File: my_enumerable_module.rb (hunk 7 above)
Comment (slabounty) [reply my_enumerable_module.rb:1]: Oh ... this is fine for the project. In real life though, you'd want to think long and hard about opening up something like this and monkey patching (term for changing an existing module class). I will note, that Rails, does this quite a bit, but DHH is probably a bit smarter than us lol!

File: spec/my_enumerable_module_spec.rb (hunk 9 above)
Comment (slabounty) [reply spec/my_enumerable_module_spec.rb:39]: Well ... one way is to search for a regex. A typical regex for an editor for an empty line or a line with only blanks might be something like ...
```
/^\s*$/
```

So `^` start of line, `\s` is a space, `*` is 0 or more, and `$` is end of line. You should get all of them that way I think.

File: my_enumerable_module.rb (hunk 7 above)
Comment (janessatran) [reply my_enumerable_module.rb:1]: Ah, gotcha. Haha he definitely knows way more than me that's for sure 😂

File: spec/my_enumerable_module_spec.rb (hunk 9 above)
Comment (janessatran) [reply spec/my_enumerable_module_spec.rb:39]: Oh I didn't know you can search for a regular expression, thank you! 👍🏻

**Comment by botanical** [comment]
Do you use VS Code, @janessatran ? You can install this extension: https://marketplace.visualstudio.com/items?itemName=misogi.ruby-rubocop 🙂

**Comment by janessatran** [comment]
Yep I do, thanks @botanical!!

**Pull request merged**